protobuf/lib/src/protobuf/field_set.dart

```diff
@@ -72,7 +72,7 @@ class _FieldSet {
   /// If [_frozenState] contains a boolean, the hashCode hasn't been memoized,
   /// so it will return null.
   int? get _memoizedHashCode =>
-      _frozenState is int ? _frozenState as int : null;
+      _frozenState is int ? downcastUnchecked<int?>(_frozenState) : null;
```

Comment:
This line makes the kernel worse, probably because of TFA results getting worse because of the function call introduced. (the call is inlined, but TFA runs before inlinings)